turtle_graphics/index.js

```diff
@@ -0,0 +1,300 @@
+/**
+ * Author: Kevin Njunge
+ * Assignment: Turtle Graphics
+ * Due Date: 3/22/2019
+ * 
+ * 
+ * 
+ * '-' => empt mark
+ * '@' => turtle foot mark
+ * 
+ */
+class Turtle{
+    constructor(x,y){
+        this.x = x;
+        this.y = y;
+        this.x_direction = 1; // move east by default
+        this.y_direction = 0;
+        this.turtle_tracks = [[x,y]]
+        this.max_x_coordinate=0,this.min_x_coordinate=0, this.max_y_coordinate=0,this.min_y_coordinate =0
+        this.get_prior_coordinates = [x,y]
+    }
+
+    updateXCoordinateIfNegative(){
+        if(this.min_x_coordinate <0){
+            const remainder = 0 - this.min_x_coordinate
+            this.min_x_coordinate +=remainder
+            this.max_x_coordinate +=remainder
+            this.turtle_tracks.forEach(function(element){
+                element[0] +=remainder
+            })
+        }
+        return this;
+    }
+    updateYCoordinateIfNegative(){
+        if(this.min_y_coordinate <0){
+            const remainder = 0 - this.min_y_coordinate
+            this.min_y_coordinate +=remainder
+            this.max_y_coordinate +=remainder
+            this.turtle_tracks.forEach(function(element){
+                element[1] +=remainder
+            })
+        }
+        return this;
+    }
+    // update list of turtle tracks with latest coordinates
+    generateTurtleTracks(x_direction,continued_point,latest_point){
+        let update_x,update_y=0
+        for(let i = continued_point;i< latest_point; i++){
+            update_x = x_direction ? i : this.x
+            update_y = x_direction ? this.y : i
+            this.turtle_tracks.push([update_x,update_y])
+        }
+        return this;
+    }
+    updateTurtleTrack(){
+        if(this.get_prior_coordinates[0]!==this.x){
+            if(this.get_prior_coordinates[0]<this.x){
+                this.generateTurtleTracks(true,(this.get_prior_coordinates[0]+1),(this.x+1))
```

Comment:
To keep code `DRY`, use constants, eg. class variables of `this.WEST` or a function `this.currentDrection = { code... }`